test/unit/postprocessing/test_threshold_optimization.py

```diff
@@ -150,24 +150,19 @@ def test_threshold_optimization_different_input_lengths(data_X_y_sf, constraints
                                    sensitive_features=data_X_y_sf.sensitive_features)
 
 
-@pytest.mark.parametrize("score_transform", candidate_Y_transforms)
 @pytest.mark.parametrize("y_transform", candidate_Y_transforms)
 @pytest.mark.parametrize("sensitive_features_transform", candidate_A_transforms)
-def test_threshold_optimization_demographic_parity(score_transform, y_transform,
+def test_threshold_optimization_demographic_parity(y_transform,
                                                    sensitive_features_transform):
     y = y_transform(labels_ex)
     sensitive_features = sensitive_features_transform(sensitive_features_ex1)
-    scores = score_transform(scores_ex)
-    estimator = ThresholdOptimizer(estimator=ExamplePredictor(scores),
+    estimator = ThresholdOptimizer(estimator=PassThroughPredictor(),
                                    constraints=DEMOGRAPHIC_PARITY)
-    estimator.fit(X_ex, y, sensitive_features=sensitive_features)
-
+    estimator.fit(pd.DataFrame(scores_ex), y, sensitive_features=sensitive_features)
```

Comment:
Why was this required? Did we break something about supported data types?

Reply:
This is because I'm now working with `PassThroughPredictor` instead of `ExamplePredictor`.

Reply:
This goes back to the same point as in the other comment. If we allow more data types we should test the entire matrix of data types. If we don't allow it we need to adjust the places where we've said we do.

Reply:
This is a different issue. The feature matrix should always be a pass-through, we don't touch it, we just pass it on to the base estimator. We cannot guarantee that the base estimator will work well with one or another kind of feature encoding (this is what burnt us with LightGBM). Maybe best is to remove all of the validation that we do around X. But that should be a separate PR.

Reply:
Wow I completely confused myself... this is passed as the feature matrix, of course, not as scores. 

What burnt us with LightGBM was that we changed X (and that was in reductions, not postprocessing). That's not the case here, though. You should be able to pass in any kind of format that we allow for X, and therefore we should also be testing any kind of format as input.

Reply:
The previous version was only trying different transformed versions of `y`, `scores_ex` and `sensitive_features`, so I've retained that functionality. Let me know if you think I should still be changing anything here.